connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaConnectorConsumer.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.kafka.inbound;
+
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.convertConsumerRecordToKafkaInboundMessage;
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.getKafkaProperties;
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.getOffsets;
+
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.impl.ConnectorInputException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaConnectorConsumer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaConnectorConsumer.class);
+
+  private InboundConnectorContext context;
+
+  public CompletableFuture<?> future;
+
+  Consumer<String, String> consumer;
+
+  boolean shouldLoop = true;
```

Comment:
Could you please explain why you chose default modifier?

Reply:
This way I am able to modify the value from tests. So the only reason is test purposes.